utils/prof_utils.py

```diff
@@ -0,0 +1,1177 @@
+def analyse_pulse_prof(prof_path=None, prof_data=None, period=None, alpha=2):
+    """
+    Estimates the signal to noise ratio from a pulse profile. Returns are in list form. Will return more for verbose==True setting, explained below.
+    NOTE: user must supply EITHER a betprof path OR prof_data and period of the pulse profile.
+    Based on code oringally writted by Nick Swainston.
+
+    Parameters:
+    -----------
+    prof_path: string
+        OPTIONAL - The path of the bestprof file
+    prof_data: list
+        OPTIONAL - A list of floats that contains the pulse profile
+    period: float
+        OPTIONAL - The pulsar's period in ms
+    verbose: boolean
+        OPTIONAL - Determines whether to return more detailed information. Detailed below
+
+    Returns:
+    --------
+    sn: float
+        The estimated signal to noise ratio
+    u_sn: float
+        The estimated signal to noise ratio's its uncertainty
+    flags: list
+        VERBOSE - a list of flagged data points
+    w_equiv_bins: float
+        VERBOSE - the equivalent width of the profile measured in bins
+    u_w_equiv_bins: float
+        VERBOSE - the uncertaintiy in w_equiv_bins
+    w_equiv_ms: float
+        VERBOSE - the equivalent width of the profile measured in ms
+    u_w_equiv_ms: float
+        VERBOSE - the uncertainty in w_equiv_ms
+    scattered: boolean
+        VERBOSE - when true, the profile is highly scattered
+    """
+    if prof_path is None and (prof_data is None or period is None):
+        logger.warning("Insufficient information to attain SN estimate from profile. Returning Nones")
+        return None, None
+
+    if prof_data is None:
+        _, _, _, period, _, _, _, prof_data, nbins = get_from_bestprof(prof_path)
+        nbins = float(nbins)
+        period = float(period)
+    else:
+        nbins = len(prof_data)
+
+    #centre the profile around the max
+    shift = -int(np.argmax(prof_data))+int(nbins)//2
+    prof_data = np.roll(prof_data, shift)
+
+    #find std and check if profile is scattered
+    sigma, flags = sigmaClip(prof_data, tol=0.01, ntrials=100, alpha=alpha)
+    bot_prof_min = (max(prof_data) - min(prof_data)) * .1 + min(prof_data)
+    scattered=False
+    if (np.nanmin(flags) > bot_prof_min) or ( not np.isnan(flags).any() ):
+        logger.warning("The profile is highly scattered. S/N estimate cannot be calculated")
+        scattered=True
+        #making a new profile with the only bin being the lowest point
+        prof_min_i = np.argmin(prof_data)
+        flags = []
+        for fi, _ in enumerate(prof_data):
+            if fi == prof_min_i:
+                flags.append(prof_data[fi])
+            else:
+                flags.append(np.nan)
+        flags = np.array(flags)
+        prof_data -= min(prof_data)
+    else:
+        prof_e = 500. #this is an approximation
+        non_pulse_bins = 0
+        #work out the above parameters
+        for i, data in enumerate(prof_data):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unused variable 'data'](https://app.codacy.com/manual/CIRA-Pulsars-and-Transients-Group/vcstools/pullRequest?prid=4645369)